3rd_party/pull-eigen.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+git subtree pull --prefix 3rd_party/eigen git@github.com:eigenteam/eigen-git-mirror.git branches/3.3 --squash
```

Comment:
I think it would be better if this command used the tag `3.3.7` rather than just getting the latest commit on the `branches/3.3` branch.

The 3.3 branch has progressed quite a lot since 3.3.7.  If you really did use `branches/3.3` to create the subtree then we're not using 3.3.7, but some subsequent commit in between 3.3.7 and 3.3.8.  I'd prefer to use the actual 3.3.7 release.

Reply:
> The Windows test failure is:
> 
> ```
> CBayesianOptimisationTest.cc(315): fatal error: in "CBayesianOptimisationTest/testMaximumExpectedImprovement": critical check maths::CBasicStatistics::mean(gain) > 1.27 has failed [1.2653069409502808 <= 1.27]
> ```
> 
> This doesn't show up nicely in Jenkins because this PR branch must branch from master after I did the first Boost.Test change, but before my subsequent PRs that fixed the Jenkins integration. So you'll get nicer Jenkins test reports if you merge latest master into your PR branch.

Wish I'd read this comment earlier - it took ages to track this one down!